multiqc/modules/biscuit/biscuit.py

```diff
@@ -0,0 +1,688 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from BISCUIT """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+import re
+
+from multiqc import config
+from multiqc.plots import beeswarm, linegraph, bargraph, table
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+def _uniquify_ID(data, header, namespace):
+    header2 = dict([(namespace+k, v) for k, v in header.items()])
+    data2 = {}
+    for sid, datum in data.items():
+        data2[sid] = dict([(namespace+k,v) for k, v in datum.items()])
+
+    return (data2, header2)
+
+# Log parsing regexes
+class MultiqcModule(BaseMultiqcModule):
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='BISCUIT', anchor='biscuit',
+        href="https://github.com/zwdzwd/biscuit",
+        info="is a tool to map bisulfite converted sequence reads and determine"\
+        " cytosine methylation states.")
+
+        # Set up data structures
+        self.mdata = {
+            'mapq': {},
+            'markdup': {},
+            'coverage': {},
+            'retention': {},
+        }
+
+        # Find and parse bismark alignment reports
+        for k in self.mdata:
+            for f in self.find_log_files('biscuit/%s' % k):
+                d = getattr(self, 'parse_logs_%s' % k)(f['f'], f['fn'])
+                self.mdata[k][f['fn']] = d
+
+        num_parsed = 0
+        for k in self.mdata:
+            num_parsed += len(self.mdata[k])
+        if num_parsed == 0:
```

Comment:
A slightly more condensed version could be:

```python
if sum([len(self.mdata[k]) for k in self.mdata]) == 0:
```

Reply:
Thanks for showing me. This version is used.